### DISCUSSION TITLE
Export support for KdenLive

### DISCUSSION THREAD

**Discussion by lexrupy** [WyattBlue/auto-editor#539] [2. Feature Ideas]
Hello, since the kdenlive project is a sort of a mlt file (like shotcut) it would be nice the exported file be compatible with kdenlive.

Unfortunattely at the point I can´t open a  shotcut timeline .mlt directly on kdenlive, but since kdenlive have way more features than shotcut me and many other people centaintly will appreciate this feature.

Thanks in advance.

**Comment by rowan-fan** [comment]
Will auto editor support exporting the editing files used by Kdenlive? I really need this feature!

This feature would be incredibly useful for streamlining my video editing workflow.

Auto editor is a powerful tool for automatic video editing, while Kdenlive is a popular open-source non-linear video editor. The ability to export project files from auto editor that can be directly imported into Kdenlive would create a seamless bridge between automated editing and manual fine-tuning.

**Comment by RobinPenn974** [comment]
What are the differences between Shotcut and Kdenlive's MLT files? Why aren't they interchangeable?

**Reply by thursdaybw** [reply]
https://docs.kdenlive.org/en/project_and_asset_management/file_management/project_files.html 

> Kdenlive’s .kdenlive project files use an XML format based on MLT’s format to describe the source media used in a project, as well as the use of that media in the timeline. For more details see [MLT's XML documentation](https://www.mltframework.org/docs/mltxml/) and [MLT's DTD](https://github.com/mltframework/mlt/blob/master/src/modules/xml/mlt-xml.dtd) (document type definition).
> 
> For most media, such as video, audio, and images, Kdenlive stores only a reference in a project, not the media itself. Only some media gets stored directly inside Kdenlive’s project files, most notably Kdenlive [title](https://docs.kdenlive.org/en/project_and_asset_management/project_bin/title_clip.html) and [color clips](https://docs.kdenlive.org/en/project_and_asset_management/project_bin/color_clip.html).
> 
> The advantages of using this file format are:
> 
>     [MLT](https://www.mltframework.org/) is able to directly render Kdenlive project files. MLT simply ignores all the additional Kdenlive-specific project data and just sticks to its rendering information. The Kdenlive-specific data is the additional icing on top that makes working with projects much easier than editing at the (lower) rendering level.
> 
>     Kdenlive can directly include and work with MLT rendering files, just the same way it works with other media. In fact, Kdenlive’s [library clips](https://docs.kdenlive.org/en/project_and_asset_management/library.html) are simply MLT rendering files, nothing more.
> 
> The project file holds all relevant information about
> 
>     target video and audio properties (selected in the [project profile](https://docs.kdenlive.org/en/project_and_asset_management/project_settings/general_settings.html))
> 
>     references to all the source materials (and to their [proxies](https://docs.kdenlive.org/en/glossary.html#term-Proxy))
> 
>     position, duration and edits of the clips in the timeline, with applied effects and their respective parameters including keyframes, and everything to get the final result
> 
> Project files are associated with a [working directory](https://docs.kdenlive.org/en/project_and_asset_management/file_management/folder_structure.html), in which Kdenlive will generate proxies and thumbnails, so that an overview of your media always shows up quickly.

More details information about the kdenlive file format internals is available here: https://github.com/KDE/kdenlive/blob/master/dev-docs/fileformat.md

**Comment by ManticorePrivacy** [comment]
Hi everyone,  

First, thank you for your work on Auto-Editor. It's been a great tool for simplifying video editing tasks.  

I’m currently exploring how Auto-Editor exports MLT XML for Shotcut and how it might align or differ with Kdenlive’s MLT implementation. From what I’ve observed:  

1. Kdenlive and Shotcut both use MLT XML as a base, but each adds editor-specific properties to the `<mlt>` tag. For example:  
   - Shotcut: `<mlt LC_NUMERIC="C" version="7.9.0" title="Shotcut version 22.09.23" producer="main_bin">`  
   - Kdenlive: `<mlt LC_NUMERIC="C" producer="main_bin" root="/home/moniika/Downloads" version="7.28.0">`  

   I’m curious how Shotcut determines the path to the video file since it’s not explicitly stated in the XML. For instance, I exported a project using:  
   ```
   auto-editor ./Downloads/input.mp4 --export shotcut
   ```
   The output XML references the video as `input.mp4` but doesn’t include a full path. My assumption is that Shotcut assumes the video is in the same directory as the XML file. Could you confirm if that’s correct ?

I’d appreciate any insights you can provide to help me better understand how Auto-Editor integrates with these tools.

Thank you for your time, and keep up the amazing work!  

Best regards

**Comment by dr04e606** [comment]
I was hoping to try contributing this feature myself, but with my very limited programming knowledge, I don’t think I’d be able to pull it off. That said, I really hope KDEnlive support gets added soon - it's a bit disappointing not to have export support for one of the best (if not the best) FOSS video editors out there.

**Comment by guladam** [comment]
Hey everyone,
I'm really much an advocate for kdenlive and I was a bit disappointed when I saw there was no export option for the app. No guarantees but I started working on an implementation, hopefully I can suceed. That would be my first ever OSS contribution so I'm stoked!

**Reply by thursdaybw** [reply]
Do you have a public repo.. or a fork with a diff.. even if it's not even close to working, it's often wise to publish it anyway.. Never know who might see it and have help to offer.. and existing code to add.

**Reply by guladam** [reply]
Thanks for the tip!
I created a fork and made my first ever pull request. 🥳

[Fork](https://github.com/guladam/auto-editor)
Pull request: #790

**Comment by guladam** [comment]
Hi everyone,

If anyone's interested, kdenlive export has been merged into the master branch!
There are some limitations to it, but it works for now.

For more details check #790

**Reply by thursdaybw** [reply]
wonderful, great job!

**Comment by WyattBlue** [comment]
Thanks to @guladam, kdenlive export is now available in 28.1.0
